spec/views/hq/sponsors/_form_spec.rb

```diff
@@ -24,129 +23,95 @@ def render_sponsor_form current_sponsor
 
-      assert_select "select#sponsor_organization_id" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "option", value: sponsor_full.organization_id
-        if sponsor_full.organization
-          assert_select "option",
-            html: CGI::escape_html(sponsor_full.organization.name)
-        else
-          assert_select "option", value: false
-        end
+      if sponsor_full.organization
+        expect(rendered).to have_selector("select[id='sponsor_organization_id'][disabled] option[selected]", text: sponsor_full.organization.name)
+      else
+        expect(rendered).to have_selector("select[id='sponsor_organization_id'][disabled] option", text: "")
+        expect(rendered).to_not have_selector("select[id='sponsor_organization_id'][disabled] option[selected]")
       end
 
-      assert_select "select#sponsor_branch_id" do
-        assert_select "[disabled=?]", "disabled"
-        assert_select "option", value: sponsor_full.branch_id
-        if sponsor_full.branch
-          assert_select "option",
-            html: CGI::escape_html(sponsor_full.branch.name)
-        else
-          assert_select "option", value: false
-        end
+      if sponsor_full.branch
+        expect(rendered).to have_selector("select[id='sponsor_branch_id'][disabled] option[selected]", text: sponsor_full.branch.name)
+      else
+        expect(rendered).to have_selector("select[id='sponsor_branch_id'][disabled] option", text: "")
+        expect(rendered).to_not have_selector("select[id='sponsor_branch_id'][disabled] option[selected]")
       end
 
-      assert_select "select#sponsor_payment_plan" do
-        assert_select "option", value: sponsor_full.payment_plan,
-                                html: CGI::escape_html(sponsor_full.payment_plan) do
-        end
-      end
+      expect(rendered).to have_select("sponsor_payment_plan", selected: sponsor_full.payment_plan,
+                                       with_options: Sponsor::PAYMENT_PLANS)
 
-      assert_select "select#sponsor_country" do
-        assert_select "option", value: sponsor_full.country do
-          assert_select "[selected]", html: CGI::escape_html(en_ar_country(sponsor_full.country).strip)
-        end
-      end
+      expect(rendered).to have_select("Country", selected: en_ar_country(sponsor_full.country))
 
-      assert_select "select#sponsor_city" do
-        assert_select "option", value: CGI::escape_html(sponsor_full.city),
-                                html: CGI::escape_html(sponsor_full.city)
-      end
+      expect(rendered).to have_select("City", selected: sponsor_full.city)
 
-      assert_select "select#sponsor_agent_id" do
-        assert_select "option", value: sponsor_full.agent_id
-        if sponsor_full.agent
-          assert_select "option",
-            html: CGI::escape_html(sponsor_full.agent.user_name)
-        else
-          assert_select "option", value: false
-        end
+      if sponsor_full.agent
```

Comment:
Agent assignment is now mandatory for all sponsors, no need for the conditional.